actors/builtin/storage_miner/storage_miner_actor.go

```diff
@@ -31,6 +31,71 @@ const epochUndefined = abi.ChainEpoch(-1)
 
 type StorageMinerActor struct{}
 
+type CronEventType int64
+
+const (
+	CronEventType_Miner_SurpriseExpiration = CronEventType(1)
+	CronEventType_Miner_WorkerKeyChange    = CronEventType(2)
+	CronEventType_Miner_PreCommitExpiry    = CronEventType(3)
+	CronEventType_Miner_SectorExpiry       = CronEventType(4)
+	CronEventType_Miner_TempFault          = CronEventType(5)
+)
+
+type CronEventPayload struct {
+	EventType CronEventType
+	Sectors   []abi.SectorNumber // Empty for global events, such as SurprisePoSt expiration.
+}
+
+///////////////////////
+// Worker Key Change //
+///////////////////////
+
+func (a *StorageMinerActor) StageWorkerKeyChange(rt Runtime, newKey addr.Address) *vmr.EmptyReturn {
+	var st StorageMinerActorState
+	rt.State().Transaction(&st, func() interface{} {
+		rt.ValidateImmediateCallerIs(st.Info.Owner)
+
+		// must be BLS since the worker key will be used alongside a BLS-VRF
+		if newKey.Protocol() != addr.BLS {
+			rt.Abort(exitcode.ErrIllegalArgument, "Worker Key must be BLS.")
+		}
+
+		keyChange := WorkerKeyChange{
+			NewWorker:   newKey,
+			EffectiveAt: rt.CurrEpoch() + indices.StorageMining_WorkerKeyChangeFreeze(),
+		}
+
+		// note that this may replace another pending key change
+		st.Info.PendingWorkerKey = keyChange
+		cronPayload := serde.MustSerializeParams(CronEventType_Miner_WorkerKeyChange, []abi.SectorNumber{})
```

Comment:
This is just a placeholder, but remove the empty last argument, which won't be involved.

Reply:
These are getting deserialized into a structure that expects the struct. I thought it might be necessary..?

Reply:
You might be right, until we write the real serialization code here. Fine to leave it.